src/tiled/mapdocument.cpp

```diff
@@ -335,6 +335,19 @@ void MapDocument::resizeMap(const QSize &size, const QPoint &offset, bool remove
     // TODO: Handle layers that don't match the map size correctly
 }
 
+
+void MapDocument::autocropMap()
+{
+    // Since autocrop option is allowed only for Tile Layer
+    TileLayer *tileLayer = static_cast<TileLayer*>(mCurrentLayer);
+
+    const QRect bounds = tileLayer->computeAutocropRegion().boundingRect();
```

Comment:
I don't see the need for a special `computeAutocropRegion` function. Just use `tileLayer->region().boundingRect()`.